integrated_channels/sap_success_factors/client.py

```diff
@@ -320,14 +320,24 @@ def _call_search_students_recursively(self, sap_search_student_url, all_inactive
             return None
 
         if 'error' in sap_inactive_learners:
-            LOGGER.warning(
-                'SAP searchStudent API for customer %s and base url %s returned response with '
-                'error message "%s" and with error code "%s".',
-                self.enterprise_configuration.enterprise_customer.name,
-                self.enterprise_configuration.sapsf_base_url,
-                sap_inactive_learners['error'].get('message'),
-                sap_inactive_learners['error'].get('code'),
-            )
+            try:
+                LOGGER.warning(
+                    'SAP searchStudent API for customer %s and base url %s returned response with '
+                    'error message "%s" and with error code "%s".',
+                    self.enterprise_configuration.enterprise_customer.name,
+                    self.enterprise_configuration.sapsf_base_url,
+                    sap_inactive_learners['error'].get('message'),
+                    sap_inactive_learners['error'].get('code'),
+                )
+            except AttributeError:
+                LOGGER.warning(
```

Comment:
looks like this fixes the immediate issue. Is this because sap_inactive_learners sometimes does not contain `error` and `code` attributes?

Reply:
I specified my findings in the description, but afaict certain error conditions are not providing this dictionary format. They updated error handling, so it is possible they missed certain errors or something. I'm not sure. They also occasionally spit back garbled Java, but we're already handling a return of complete nonsense.

Reply:
So to answer what I think you may be asking about it not returning a dictionary in general, no, it definitely does. It's that sometimes the _subdictionary_ in the error key is missing.